tests/TestCase.php

```diff
@@ -8,14 +8,16 @@
 use Adldap\Query\Builder;
 use Adldap\Query\Grammar;
 use Adldap\Connections\ConnectionInterface;
-use Mockery\Adapter\Phpunit\MockeryTestCase;
+use LDAP\Result;
 
-class TestCase extends MockeryTestCase
+class TestCase extends \PHPUnit\Framework\TestCase
```

Comment:
Is there a reason we've changed the test case back to a PHPUnit framework test case? Wouldn't this cause mockery to not shut down properly per test?

Reply:
The *void* return type has been introduced from PHP 7.1 so I had to remove it from *TestCase* to make it compatible with PHP 7.0.
Unfortunately [MockeryTestCase](https://github.com/mockery/mockery/blob/4969eea/library/Mockery/Adapter/Phpunit/MockeryTestCase.php) implements the methods [`setUp()` and `tearDown()`](https://github.com/mockery/mockery/blob/7a02141/library/Mockery/Adapter/Phpunit/MockeryTestCaseSetUp.php#L27) with *void* as a return type, so I got this error: `Declaration of TestCase::setUp() must be compatible with MockeryTestCase::setUp(): void`
Here a simplified example: https://3v4l.org/1Gje5

About Mockery'shutdown, I only had to modify `tearDown()` by adding this call: `$this->addToAssertionCount(Mockery::getContainer()->mockery_getExpectationCount());`
If you take a look at [mockeryTestTearDown()](https://github.com/mockery/mockery/blob/4969eea/library/Mockery/Adapter/Phpunit/MockeryTestCase.php#L32) you'll see that the method is empty, so we just lose the ability to hook up on this.
There is still a slighty difference about [how exceptions are handled](https://github.com/mockery/mockery/blob/4969eeac6afbeb99e74c0ad323e6bb3c3838e6f4/library/Mockery/Adapter/Phpunit/MockeryPHPUnitIntegration.php#L53) but I don't think this is used by this project.

Reply:
Ok understood. So we don't have to call `Mockery::close()` either?

https://github.com/mockery/mockery/blob/4969eeac6afbeb99e74c0ad323e6bb3c3838e6f4/library/Mockery/Adapter/Phpunit/MockeryPHPUnitIntegration.php#L66-L70

Reply:
Well, I think we should actually.
I'll have to do some tests to see how to integrate it and then I'll modify the PR.

Reply:
Ok sounds good! Thanks again for your hard work on this ❤️

Reply:
I've updated the PR to have the same behaviour as before as possible.

Reply:
@stevebauman Does it look right to you now?

Reply:
Looks good to me @villfa!